docs/resource-tuning-recommendations.md

```diff
@@ -0,0 +1,122 @@
+- **15-25% of CPU Requests** on the platform are related to Jenkins
+- **7% of the CPU Requests** are actually used, on average, over a 1 Day period
+- **10% or More CPU Requests** for the overall platform can be saved by tuning Jenkins resources 
+
+### Recommended Configuration
+Based on the performance testing details below, the following recommendations are suggested for Jenkins deployments: 
+- CPU Request: 100m
+- CPU Limit: 1000m (May vary depending on usage)
+- Memory Request: 512M
+- Memory Limit: 1-2GB (May vary depending on usage)
+
+On a typical Jenkins deployment, the following snippet could be used if you are editing the yaml: 
+```
+        resources:
+          requests:  
+            cpu: "100m"
+            memory: "128Mi"
+          limits:
+            cpu: "1"
+            memory: "512Mi"
+```
+
+The following command can also be used to update a Jenkins DeploymentConfig: 
+
+```
+oc patch dc/jenkins -p '{"spec": {"template": {"containers":{"resources":{"requests": {"cpu":"100m", "memory":"128Mi"}, "limits": {"cpu":"1", "memory":"512Mi"}}}}}}'
+```
+
+###  Performance Testing Details
+The reason that Jenkins is often deployed with such high CPU and Memory Requests was related to previous scheduler issues that have since been fixed on the platform. As a result, the templates **and existing Jenkins deployments** should be tuned to reduce the CPU requests. 
+
+A test was performed to collect the startup time of Jenkins under various resource configurations. Each test was performed 3 times and the startup time was averaged out across each iteration. The name of each test is in the format of `[cpu_requests_in_milllicores]-[cpu_limits_in_millicores]-[memory_requests_in_mb]`. 
+
+![](images/jenkins_performance_test_results.png)
+
+| Test Name                  | Average Startup Time (s) |
+|----------------------------|--------------------------|
+| 100m-req-500m-limit-128m   |                      295 |
+| 100m-req-500m-limit-512m   |                      248 |
+| 100m-req-500m-limit-128m   |                      368 |
+| 100m-req-1000m-limit-128m  |                      163 |
+| 100m-req-500m-limit-512m   |                      185 |
+| 100m-req-1000m-limit-512m  |                       77 |
+| 100m-req-2000m-limit-512m  |                       80 |
+| 500m-req-1000m-limit-128m  |                      137 |
+| 500m-req-1000m-limit-512m  |                       91 |
+| 1000m-req-2000m-limit-128m |                      131 |
+| 1000m-req-2000m-limit-512m |                       73 |
+
+The observations from the testing can be summarized as follows: 
+- CPU Limit has the largest affect on Startup Performance
+- CPU Request has little affect on Startup Performance
+- The gain from a CPU Limit of 500m to 1000m is major
+- The gain from a CPU Limit of 1000m to 2000m is minor
+- One ideal configuration looks like:
```

Comment:
Isn't this just a duplicate of the "Recommended Configuration" section above?

Reply:
Yes it is, but tied to the performance results.